torch_xla_py/xla_dist.py

```diff
@@ -33,17 +33,51 @@ def __init__(self, client_workers, service_workers):
       client_workers: a list of ClientWorker objects.
       service_workers: a list of ServiceWorker objects.
     """
+    for client_worker, service_worker in zip(client_workers, service_workers):
```

Comment:
fyi: this check will miss the case where client and service workers are of different lengths, as zip stops when one of the iterators is exhausted

Reply:
interesting case when one or both are empty, for example.